### PR TITLE
Fix caching

### PR DESCRIPTION
**Pull request by thiagomurakami**
#### What is the purpose of this pull request?

Delete headers and set vary for graphql routes according to scope.

#### What problem is this solving?

Public routes that should be cached.

#### How should this be manually tested?

???

#### Types of changes

* [x] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: src/service/http/middlewares/vary.ts [hunk 1]
```diff
@@ -1,22 +1,81 @@
...
+function isGraphqlContext <T extends IOClients, U, V> (ctx: ServiceContext<T,U,V>): ctx is ServiceContext<T,U,V> & Pick<GraphQLContext, 'graphql'> {
+  if ('graphql' in ctx) {
+    return true
+  }
+  return false
+}
+
+function setVaryIfExists<T extends IOClients, U, V> (ctx: ServiceContext<T, U, V>, varyHeaders: string[]) {
+  varyHeaders.forEach((varyHeader) => {
+    if (ctx.get(varyHeader)) {
```
Comment (brunoabreu) [inline src/service/http/middlewares/vary.ts]: We should always add `vary`, even if the header is not there.

File: src/service/http/middlewares/vary.ts [hunk 2]
```diff
@@ -1,22 +1,81 @@
...
+      ctx.vary(SEGMENT_HEADER)
+    }
+    if (ctx.get(SESSION_HEADER)) {
+      ctx.vary(SESSION_HEADER)
+    }
   }
 
   await next()
+
+  if (isGraphql && isGraphqlContext(ctx)) {
```
Comment (brunoabreu) [inline src/service/http/middlewares/vary.ts]: I'm not happy with this middleware being aware of graphql, but I can't think a better way to do it 😞

File: src/service/http/middlewares/cdnNormalizer.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    forbidden: [SESSION_HEADER],
+    path: '/_v/segment.',
+    vary: [SEGMENT_HEADER],
+  },
+]
+
+
+export async function cdnNormalizer <T extends IOClients, U, V> (ctx: ServiceContext<T, U, V>, next: () => Promise<any>) {
+  const { path } = ctx
+  const strategy = cachingStrategies.find(cachingStrategy => path.indexOf(cachingStrategy.path) === 0)
```
Comment (thiagomurakami) [inline src/service/http/middlewares/cdnNormalizer.ts]: https://jsperf.com/publicregexstartswith

File: src/service/graphql/middlewares/response.ts [hunk 4]
```diff
@@ -23,10 +21,6 @@ export async function response (ctx: GraphQLServiceContext, next: () => Promise<
     'Cache-Control': cacheControlHeader,
   })
 
-  if (segment) {
```
Comment (brunoabreu) [inline src/service/graphql/middlewares/response.ts]: 🤔

**Pull request merged**